### DISCUSSION TITLE
Web Client

### DISCUSSION THREAD

**Discussion by TomBenDor** [TomBenDor/AP2#134] [Announcements]
## What's Changed
* Created React app by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/3
* Create Node.js compile action by @TomBenDor in https://github.com/TomBenDor/AP2/pull/9
* Created basic routes by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/8
* Update node.js.yml by @TomBenDor in https://github.com/TomBenDor/AP2/pull/10
* Save user credentials by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/13
* Set action to run on Windows & fix warnings by @TomBenDor in https://github.com/TomBenDor/AP2/pull/14
* Action runs only on node 16.x by @TomBenDor in https://github.com/TomBenDor/AP2/pull/23
* Run action only when PR is ready by @TomBenDor in https://github.com/TomBenDor/AP2/pull/24
* Auth System Design by @TomBenDor in https://github.com/TomBenDor/AP2/pull/25
* Added use of CSS variables by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/31
* Better support for small screens by @TomBenDor in https://github.com/TomBenDor/AP2/pull/32
* Landing Page by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/26
* removed comment by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/37
* Upload a profile picture by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/34
* Live field errors on Sign Up by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/44
* Display name checks by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/45
* Fixed Sign Up bugs by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/46
* SVG animation alignment & scaling by @TomBenDor in https://github.com/TomBenDor/AP2/pull/47
* Additional Sign Up errors by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/48
* Display name regex bug by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/50
* Fixed .center class by @TomBenDor in https://github.com/TomBenDor/AP2/pull/52
* Sign in username checks by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/55
* Split css file by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/54
* Action runs when PR is not draft by @TomBenDor in https://github.com/TomBenDor/AP2/pull/59
* Chat Page by @TomBenDor in https://github.com/TomBenDor/AP2/pull/40
* Fixed check passes without requirements by @TomBenDor in https://github.com/TomBenDor/AP2/pull/61
* Hardcoded database by @TomBenDor in https://github.com/TomBenDor/AP2/pull/62
* Chat background image by @TomBenDor in https://github.com/TomBenDor/AP2/pull/64
* Run action when labeled by @TomBenDor in https://github.com/TomBenDor/AP2/pull/69
* Sending messages by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/68
* Chat welcome message by @TomBenDor in https://github.com/TomBenDor/AP2/pull/67
* Profile picture in chat page by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/71
* Send message when Enter key is pressed by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/73
* Support Multi line messages by @TomBenDor in https://github.com/TomBenDor/AP2/pull/74
* Introduced use of message cache by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/76
* Scroll down automatically in chat by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/78
* Fixed bug by @TomBenDor in https://github.com/TomBenDor/AP2/pull/82
* Sorting contacts list by last message time by @TomBenDor in https://github.com/TomBenDor/AP2/pull/77
* Spaces allowed by @TomBenDor in https://github.com/TomBenDor/AP2/pull/81
* Message width is limited to avoid scrolling by @TomBenDor in https://github.com/TomBenDor/AP2/pull/83
* Fixed bug with Messages cache by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/85
* Design chat buttons by @TomBenDor in https://github.com/TomBenDor/AP2/pull/79
* Changes to password validate both password fields by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/90
* Attach images by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/91
* Attaching Videos by @TomBenDor in https://github.com/TomBenDor/AP2/pull/93
* Better looking chat buttons by @TomBenDor in https://github.com/TomBenDor/AP2/pull/94
* Audio messages by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/95
* Clear input box after sending text message by @TomBenDor in https://github.com/TomBenDor/AP2/pull/102
* Fixed time bug by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/103
* Adding contacts by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/98
* Clear input field after adding contact by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/107
* Added discard recording button by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/108
* Initialize message box value by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/109
* Move database to JSON file by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/113
* Add more example contacts by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/97
* Add contact ux changes by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/116
* Implement dark and light theme button by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/117
* Import Lato Font by @TomBenDor in https://github.com/TomBenDor/AP2/pull/119
* Restructre database by @TomBenDor in https://github.com/TomBenDor/AP2/pull/118
* Changed message bubble colors by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/122
* Database typo by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/123
* Fixed "don't yell at me" bug by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/125
* Fixed database id bug by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/127
* Unread background color by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/126
* Contact modal ui tweaks by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/124
* Using Lato Font 300 & 700 by @TomBenDor in https://github.com/TomBenDor/AP2/pull/130
* Add more conversations to the database by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/129
* Passwords match regex pattern by @OmerShoulstein in https://github.com/TomBenDor/AP2/pull/131
* Add "Take the L" video message by @Tom-stack3 in https://github.com/TomBenDor/AP2/pull/133
* Web Client README by @TomBenDor in https://github.com/TomBenDor/AP2/pull/92

**Full Changelog**: https://github.com/TomBenDor/AP2/commits/web-client

<hr /><em>This discussion was created from the release <a href='https://github.com/TomBenDor/AP2/releases/tag/web-client'>Web Client</a>.</em>

**Comment by Tom-stack3** [comment]
Hurray! :partying_face::partying_face::partying_face:

**Comment by OmerShoulstein** [comment]
🎉🎉🎉
